File: csharp/triangle/Triangle.cs
using System;
using System.Linq;
using System.Collections.Generic;

public static class Triangle
{
    public static bool IsScalene(double side1, double side2, double side3)
    {
		var sides = new HashSet<double>(){ side1, side2, side3 };
        return sides.Count == 3 && !sides.Contains(0) && IsTriangle(side1, side2, side3);
    }

    public static bool IsIsosceles(double side1, double side2, double side3) 
    {
        var sides = new HashSet<double>(){ side1, side2, side3 };
        return sides.Count <= 2 && !sides.Contains(0) && IsTriangle(side1, side2, side3);
    }

    public static bool IsEquilateral(double side1, double side2, double side3) 
    {
        var sides = new HashSet<double>(){ side1, side2, side3 };
        return sides.Count == 1 && !sides.Contains(0) && IsTriangle(side1, side2, side3);
    }

	public static bool IsDegenerate(double side1, double side2, double side3) 
	{
		double[] sides = { side1, side2, side3 };
		var isDegenerate = (side1 + side2) == side3 || (side1 + side3) == side2 || (side2 + side3) == side1;
        return isDegenerate && !sides.Contains(0) && IsTriangle(side1, side2, side3);
	}

    private static bool IsTriangle(double side1, double side2, double side3)
	{
		return (side1 + side2) >= side3 && (side1 + side3) >= side2 && (side2 + side3) >= side1;
	}
}

File: javascript/triangle/triangle.js
//
// This is only a SKELETON file for the 'Triangle' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Triangle {
  constructor(sideA, sideB, sideC) {
    this.sideA = sideA;
    this.sideB = sideB;
    this.sideC = sideC;
  }

  isEquilateral() {
    return this.isValidTriangle() && this.sideA === this.sideB && this.sideA === this.sideC;
  }

  isIsosceles() {
    return this.isValidTriangle() && this.calculateUniqueSides() <= 2;
  }

  isScalene() {
    return this.isValidTriangle() && this.calculateUniqueSides() === 3;
  }

  isValidTriangle() {
    return this.checkInequality() && this.sideA > 0 && this.sideB > 0 && this.sideC > 0;
  }

  calculateUniqueSides() {
    let sides = new Set();
    sides.add(this.sideA).add(this.sideB).add(this.sideC);
    return sides.size; 
  }

  checkInequality() {
    return (this.sideA + this.sideB) >= this.sideC && (this.sideA + this.sideC) >= this.sideB && (this.sideB + this.sideC) >= this.sideA;
  }

  checkDegenerate() {
    return (this.sideA + this.sideB) == this.sideC || (this.sideA + this.sideC) == this.sideB || (this.sideB + this.sideC) == this.sideA;
  }
}
